src/main/java/io/tus/java/client/TusClient.java

```diff
@@ -52,6 +54,24 @@ public URL getUploadCreationURL() {
         return uploadCreationURL;
     }
 
+    /**
+     * Set the upload proxy that will be used with the {@link #uploadCreationURL}.
```

Comment:
```suggestion
     * Set the upload proxy that will be used for all requests.
```

The comment suggests that it is only relevant for the upload creation requests, but I think this proxy is used for all requests.